pkg/apis/apps/validation/validation.go

```diff
@@ -288,17 +289,21 @@ func ValidateControllerRevisionUpdate(newHistory, oldHistory *apps.ControllerRev
 }
 
 // ValidateDaemonSet tests if required fields in the DaemonSet are set.
-func ValidateDaemonSet(ds *apps.DaemonSet, opts apivalidation.PodValidationOptions) field.ErrorList {
+func ValidateDaemonSet(ds *apps.DaemonSet, featureGate featuregate.FeatureGate, opts apivalidation.PodValidationOptions) field.ErrorList {
 	allErrs := apivalidation.ValidateObjectMeta(&ds.ObjectMeta, true, ValidateDaemonSetName, field.NewPath("metadata"))
-	allErrs = append(allErrs, ValidateDaemonSetSpec(&ds.Spec, field.NewPath("spec"), opts)...)
+	if !featureGate.Enabled(features.SkipReadOnlyValidationGCE) {
```

Comment:
Don't need check the gate here - it's checked inside ValidateDaemonSetSpec (same in others)